packages/ember-runtime/lib/computed/reduce_computed_macros.js

```diff
@@ -45,7 +45,9 @@ function arrayMacro(dependentKey, callback) {
     } else {
       return emberA();
     }
-  }, { dependentKeys: [ dependentKey ], readOnly: true });
+  }, { readOnly: true });
+
+  cp.property(dependentKey);
```

Comment:
👍 the fix is good, but I wanted to comment here to explain why this is different.

The `ComputedProperty` constructor does **not** run the provided dependent keys through `expandProperties` (perhaps it should, but that is another issue altogether).  So moving this out from the constructor options into `cp.property` forces it to run `expandProperties`.

Reply:
I think it is intentional that constructor argument  does not run through `expandProperties` to avoid extra work (so it expects expanded properties already) because I made `dependent keys ` passable through constructor :P actually before